GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,478 @@
+
+        /// <summary>
+        /// Returns a new Region from a parameter set similar to GoRogue.Rectangle
+        /// </summary>
+        /// <param name="origin"></param>
+        /// <param name="width"></param>
+        /// <param name="height"></param>
+        /// <returns>A new region in the shape of a rectangle</returns>
+        public static Region Rectangle(Point origin, int width, int height)
+            => FromRectangle(new Rectangle(origin, origin + (width, height)));
+
+        /// <summary>
+        /// Creates a new Region from a GoRogue.Rectangle.
+        /// </summary>
+        /// <param name="r">The rectangle</param>
+        /// <returns>A new region in the shape of a rectangle</returns>
+        public static Region FromRectangle(Rectangle r)
+            => new Region(r.MinExtent, (r.MaxExtentX, r.MinExtentY), r.MaxExtent, (r.MinExtentX, r.MaxExtentY));
+
+
+         /// <summary>
+         /// Creates a new Region in the shape of a parallelogram, with diagonals going down and right.
+         /// </summary>
+         /// <param name="origin">Origin of the parallelogram.</param>
+         /// <param name="width">Width of the parallelogram.</param>
+         /// <param name="height">Height of the parallelogram.</param>
+         /// <returns>A new region in the shape of a parallelogram</returns>
+         public static Region ParallelogramFromTopCorner(Point origin, int width, int height)
+         {
+             Point nw = origin;
+             Point ne = origin + new Point(width, 0);
+             Point se = origin + new Point(width * 2, height);
+             Point sw = origin + new Point(width, height);
+
+             return new Region(nw, ne, se, sw);;
+         }
+
+         /// <summary>
+         /// Creates a new Region in the shape of a parallelogram, with diagonals going up and right.
+         /// </summary>
+         /// <param name="origin">Origin of the parallelogram.</param>
+         /// <param name="width">The horizontal length of the top and bottom sides.</param>
+         /// <param name="height">Height of the parallelogram.</param>
+         /// <returns>A new region in the shape of a parallelogram</returns>
+         public static Region ParallelogramFromBottomCorner(Point origin, int width, int height)
+         {
+             var negative = Direction.YIncreasesUpward ? 1 : -1;
+
+             Point nw = origin + (height, height * negative);
+             Point ne = origin + (height + width, height * negative);
+             Point se = origin + (width, 0);
+             Point sw = origin;
+
+             return new Region(nw, ne, se, sw);
+         }
+        #endregion
+
+        /// <summary>
+        /// Returns a string detailing the region's corner locations.
+        /// </summary>
+        public override string ToString()
+            => $"Region: NW{NorthWestCorner.ToString()}=> NE{NorthEastCorner.ToString()}=> SE{SouthEastCorner.ToString()}=> SW{SouthWestCorner.ToString()}";
```

Comment:
Minor potential cleanup: I think string interpolation does an implicit `ToString` for you, so you may be able to write this as:
```suggestion
            => $"Region: NW{NorthWestCorner}=> NE{NorthEastCorner}=> SE{SouthEastCorner}=> SW{SouthWestCorner}";
```